src/services/garbageCleaningService.js

```diff
@@ -0,0 +1,63 @@
+import api from "@api/api";
+import encryptionService from "./encryptionService";
+import indexedDB from "@store/indexedDB";
+import localforage from "localforage";
+import store from "@store/store";
+import { clearMessageIdsToLocalLimit } from "@store/values/Conversations";
+import { clearMessagesToLocalLimit } from "@store/values/Messages";
+import { setSelectedConversation } from "@store/values/SelectedConversation";
+import { setUserIsLoggedIn } from "@store/values/UserIsLoggedIn";
+import { updateNetworkState } from "@store/values/NetworkState";
+
+class GarbageCleaningService {
+  async #clearAccountData() {
+    await encryptionService.clearStoredAccount();
+  }
+
+  #clearAccountDataSync() {
+    encryptionService.clearStoredAccount();
+  }
+
+  #clearLocalMessageIds(cid) {
+    store.dispatch(clearMessageIdsToLocalLimit(cid));
+  }
+
+  #clearLocalMessages(cid) {
+    store.dispatch(clearMessagesToLocalLimit(cid));
+  }
+
+  async clearConversationMessages(cid) {
+    if (!cid) return;
+    this.#clearLocalMessageIds(cid);
+    this.#clearLocalMessages(cid);
+  }
+
+  async resetDataOnAuth() {
+    await this.#clearAccountData();
+    await this.clearAllLocalData();
+    store.dispatch(setSelectedConversation({}));
+    store.dispatch(setUserIsLoggedIn(true));
+  }
+
+  async handleLogout() {
+    await api.userLogout();
+    await this.#clearAccountData();
```

Comment:
The naming is confusing a bit, e.g.

`clearAccountData` and `clearAllLocalData`

Maybe we can have only one method:

```
 async clearStorage() {
   await Promise.all([localforage.clear(), indexedDB.removeAllMessages()]);
   await encryptionService.clearStoredAccount();
 }
```